### PR TITLE
Add Gas Grenades

### PR DESCRIPTION
**Pull request by AutumnNova**
### Notes:

Basically just a grenade with an internal tank that dumps it onto the tile its on
Very useful for rapidly refilling a room lacking atmos (or filling a room that has atmos with whatever terrible gas you want)
Filled (or refilled) by inserting into a gas canister just like you would an oxygen tank
Adds 2 empty ones that spawn in the EngiVend on roundstart
Comes with a colored overlay that changes depending on the contents of the grenade

(From left to right, Tritium, CO2, Plasma, Oxygen)

Also adds a way to prevent gas containers from being used as a source of gas for gasmasks, because you really shouldnt be hooking up a grenade to your gasmask

### Changelog:

CL: [New] Adds grenades that can be filled with any gas.

### CODE REVIEW

**Comment by AtnerNT** [comment]
Could you add the empty variant to the autolathe? Would be nice to be able to manufacture more.

**Comment by MaxIsJoe** [comment]
To add it to the autolathe, just go to one of the designs json file in StreamingAssets and add a new entry; like this:
```
 {
        "Internal_ID": "largecrossbow",
        "Description": "A reverse-engineered energy crossbow favored by syndicate infiltration teams and carp hunters.",
        "Category": [
            "Weapons"
        ],
        "Materials": {
            "Metal": 5000,
            "Glass": 1500,
            "Uranium": 1500,
            "Silver": 1500
        },
        "Machinery_type": [
            "ProtoLathe"
        ],
        "Compatible_machinery": [
            "Security"
        ],
        "name": "Energy Crossbow",
        "Item_ID": "/obj/item/weapon/gun/energy/kinetic_accelerator/crossbow/large"
    }
```

Item ID is always usually the GUID in the prefab tracker component of a prefab.

**Comment by AtnerNT** [comment]
> To add it to the autolathe, just go to one of the designs json file in StreamingAssets and add a new entry; like this:

The autolathe actually just references the prefabs directly as it isnt part of the techweb. Design files are only needed for machines that use the techweb

**Comment by MaxIsJoe** [comment]
in that case ignore that part and just add its prefab directly to it (not sure what that list is)

**Comment by AtnerNT** [comment]
ScriptableObjects/Machines/AutolatheProducts

**Comment by AutumnNova** [comment]
Costs 500 metal and 50 glass, takes 5 seconds to make

File: UnityProject/Assets/Scripts/Items/Weapons/AtmosGrenade.cs [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+			UpdateOverlay();
+			grenade.OnExpload.AddListener(ReleaseGas);
+		}
+
+		private void OnDestroy()
+		{
+			grenade.OnExpload.RemoveListener(ReleaseGas	);
+		}
+		
+		public void OnInventoryMoveServer(InventoryMove info)
```
Comment (Bod9001) [inline UnityProject/Assets/Scripts/Items/Weapons/AtmosGrenade.cs]: Shouldn't need to hook into the inventory movement code, Probably better to hook into the gas containers if there is any way of checking if it changes?

Comment (Bod9001) [reply UnityProject/Assets/Scripts/Items/Weapons/AtmosGrenade.cs]: humm doesn't look like there's an event for a gas mix changing, could add one but idk if it's worth it, 
any reviewers reading this I think it's fine to merge

Comment (AutumnNova) [reply UnityProject/Assets/Scripts/Items/Weapons/AtmosGrenade.cs]: I could add an action to GasContainer and invoke it within GasMixLocal's setter or something? I believe that would work

**Review by MaxIsJoe — APPROVED** [review]

**Pull request merged**